x-pack/platform/plugins/shared/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -1228,4 +1243,77 @@ export class AlertsClient {
       fields,
     };
   }
+
+  public getRuleTypeIds(ruleTypeIds: string | string[] | undefined): string[] {
+    // fetch all rule types if no specific rule type Ids are provided
+    if (!ruleTypeIds || ruleTypeIds.length === 0) {
+      const registeredRuleTypes = this.getRuleList();
+
+      if (!registeredRuleTypes) {
+        throw Boom.badRequest('No rule types are registered');
+      }
+
+      return Array.from(registeredRuleTypes.keys());
+    }
+
+    return Array.isArray(ruleTypeIds) ? ruleTypeIds : [ruleTypeIds];
+  }
+
+  public async getAlertFields(
+    ruleTypeIds: string | string[]
+  ): Promise<{ alertFields: BrowserFields; fields: FieldDescriptor[] }> {
+    const allRuleTypesIds = this.getRuleTypeIds(ruleTypeIds);
+
+    const authorizedRuleTypes = await this.authorization.getAllAuthorizedRuleTypesFindOperation({
+      authorizationEntity: AlertingAuthorizationEntity.Alert,
+      ruleTypeIds: allRuleTypesIds,
+    });
+    const authorizedRuleTypesIds = Array.from(authorizedRuleTypes.keys());
+
+    const [siemRuleTypeIds, otherRuleTypeIds] = partition(authorizedRuleTypesIds, (ruleTypeId) =>
+      isSiemRuleType(ruleTypeId)
+    );
+
+    const [siemIndices, otherIndices] = await Promise.all([
+      siemRuleTypeIds ? this.getAlertIndicesAlias(siemRuleTypeIds) : [],
+      otherRuleTypeIds ? this.getAlertIndicesAlias(otherRuleTypeIds) : [],
+    ]);
+
+    const indexPatternsFetcherAsInternalUser = new IndexPatternsFetcher(this.esClient);
+    const indexPatternsFetcherAsScoped = this.esClientScoped
+      ? new IndexPatternsFetcher(this.esClientScoped)
+      : undefined;
+
+    const [{ fields: specFields }, { fields: descriptorFields }] = await Promise.all([
+      // Fetch fields for SIEM indices using current user client
+      siemIndices.length && indexPatternsFetcherAsScoped
+        ? await indexPatternsFetcherAsScoped.getFieldsForWildcard({
+            pattern: siemIndices.join(','),
+            metaFields: ['_id', '_index'],
+            fieldCapsOptions: { allow_no_indices: true },
+            includeEmptyFields: false,
+            indexFilter: INDEX_FILTER,
+          })
+        : { fields: [] },
+      // Fetch fields for other indices using internal user client
+      otherIndices.length
+        ? await indexPatternsFetcherAsInternalUser.getFieldsForWildcard({
+            pattern: otherIndices,
+            metaFields: ['_id', '_index'],
+            fieldCapsOptions: { allow_no_indices: true },
+            includeEmptyFields: false,
+            indexFilter: INDEX_FILTER,
+          })
+        : { fields: [] },
+    ]);
+
+    const uniqueFields = mergeUniqueFieldsByName(descriptorFields, specFields);
+
+    const mappedFields = {
+      alertFields: fieldDescriptorToBrowserFieldMapper(uniqueFields),
```

Comment:
When working on my last OnWeek I noticed this in the `browser_fields` API and thought it was quite a waste of resources to duplicate the return value in two different format. Since this is a new endpoint and we wouldn't risk introducing breaking changes, can we take the occasion and remove this duplication? I'd be happy to help by sharing what I did in my PoC 🙂

Reply:
Its about returning fields similar to discover, right? 
We are planning to use this API in UI in maintenance window and replace `browser_fields` gradually in future.  But it would be nice to discuss the details and if its feasible or not. Wdyt @cnasikas ?

Reply:
Oh no it's just about the duplicated format: the old API returned the same exact fields in two ways (a list and a nested object grouped by ECS namespace). It's a huge waste of resources imo, we could just choose one of the two and group/flatten on the frontend if needed. I tried this in my PoC for the discover-like fields picker and it worked well with returning just the nested format.

Reply:
hmm.. makes sense. Which format do you prefer? I remember few of UI hooks expect array of fields.

Reply:
In my PoC I kept only the nested one and reconstructed the flattened list on the fe if necessary, but I I was mainly focusing on quickly adding ECS metadata to fields and groups. If we ever want to do something similar we could easily move the ECS groups part in a single, smaller API call to avoid repeating the metadata each time we fetch the fields. So perhaps only returning the list is the best option here?

Reply:
I agree with @umbopepato. Let's return the format we will use. Based on my understanding the KQL bars thus the data views using only the `fields` and the fields constructed by `fieldDescriptorToBrowserFieldMapper` is not needed. I would suggest what Umbo said, returning only the fields but in a format like `{ fields }` so we can extend it in the future if needed.